alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexWorkflowAction.kt

```diff
@@ -0,0 +1,559 @@
+                    } else {
+                        // rolesToRemove: these are the backend roles to remove from the monitor
+                        val rolesToRemove = user.backendRoles - request.rbacRoles.orEmpty()
+                        // remove the monitor's roles with rolesToRemove and add any roles passed into the request.rbacRoles
+                        val updatedRbac = currentWorkflow.user?.backendRoles.orEmpty() - rolesToRemove + request.rbacRoles.orEmpty()
+                        request.workflow = request.workflow.copy(
+                            user = User(user.name, updatedRbac, user.roles, user.customAttNames)
+                        )
+                    }
+                } else {
+                    request.workflow = request.workflow
+                        .copy(user = User(user.name, currentWorkflow.user!!.backendRoles, user.roles, user.customAttNames))
+                }
+                log.debug("Update monitor backend roles to: ${request.workflow.user?.backendRoles}")
+            }
+
+            request.workflow = request.workflow.copy(schemaVersion = IndexUtils.scheduledJobIndexSchemaVersion)
+            val indexRequest = IndexRequest(SCHEDULED_JOBS_INDEX)
+                .setRefreshPolicy(request.refreshPolicy)
+                .source(request.workflow.toXContentWithUser(jsonBuilder(), ToXContent.MapParams(mapOf("with_type" to "true"))))
+                .id(request.workflowId)
+                .setIfSeqNo(request.seqNo)
+                .setIfPrimaryTerm(request.primaryTerm)
+                .timeout(indexTimeout)
+
+            try {
+                val indexResponse: IndexResponse = client.suspendUntil { client.index(indexRequest, it) }
+                val failureReasons = checkShardsFailure(indexResponse)
+                if (failureReasons != null) {
+                    actionListener.onFailure(
+                        AlertingException.wrap(OpenSearchStatusException(failureReasons.toString(), indexResponse.status()))
+                    )
+                    return
+                }
+                actionListener.onResponse(
+                    IndexWorkflowResponse(
+                        indexResponse.id, indexResponse.version, indexResponse.seqNo,
+                        indexResponse.primaryTerm, request.workflow
+                    )
+                )
+            } catch (t: Exception) {
+                actionListener.onFailure(AlertingException.wrap(t))
+            }
+        }
+
+        private fun checkShardsFailure(response: IndexResponse): String? {
+            val failureReasons = StringBuilder()
+            if (response.shardInfo.failed > 0) {
+                response.shardInfo.failures.forEach { entry ->
+                    failureReasons.append(entry.reason())
+                }
+                return failureReasons.toString()
+            }
+            return null
+        }
+    }
+
+    suspend fun validateRequest(request: IndexWorkflowRequest) {
```

Comment:
Should there be another check to make sure there is only 1 input since that is what is being supported here?

Reply:
Makes sense!